test/e2e/syncer/tunnels_test.go

```diff
@@ -279,3 +281,184 @@ func TestSyncerTunnel(t *testing.T) {
 		return podLogs.Len() > 1, podLogs.String()
 	}, wait.ForeverTestTimeout, time.Millisecond*100, "couldn't get downstream pod logs for deployment %s/%s", d.Namespace, d.Name)
 }
+
+// TestSyncerTunnelFilter ensures that the syncer tunnel will reject trying to access a Pod that is crafted and not actually upsynced.
+func TestSyncerTunnelFilter(t *testing.T) {
+	t.Parallel()
+	framework.Suite(t, "transparent-multi-cluster")
+
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.SyncerTunnel, true)()
+
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	t.Cleanup(cancelFunc)
+
+	kcpServer := framework.PrivateKcpServer(t)
+	orgPath, _ := framework.NewOrganizationFixture(t, kcpServer, framework.TODO_WithoutMultiShardSupport())
+	locationPath, locationWs := framework.NewWorkspaceFixture(t, kcpServer, orgPath, framework.TODO_WithoutMultiShardSupport())
+	locationWsName := logicalcluster.Name(locationWs.Spec.Cluster)
+	userPath, userWs := framework.NewWorkspaceFixture(t, kcpServer, orgPath, framework.TODO_WithoutMultiShardSupport())
+	userWsName := logicalcluster.Name(userWs.Spec.Cluster)
+
+	// Creating synctarget and deploying the syncer
+	syncerFixture := framework.NewSyncerFixture(t, kcpServer, locationPath, framework.WithSyncedUserWorkspaces(userWs)).CreateSyncTargetAndApplyToDownstream(t)
+	syncerFixture = syncerFixture.StartAPIImporter(t)
+	syncerFixture.StartHeartBeat(t)
```

Comment:
isn't it possible to chain these function calls ?

Reply:
Done